presto-main/src/main/java/com/facebook/presto/event/QueryMonitor.java

```diff
@@ -328,10 +334,24 @@ private List<OperatorStatistics> createOperatorStatistics(QueryInfo queryInfo)
                         operatorSummary.getPeakTotalMemoryReservation(),
                         operatorSummary.getSpilledDataSize(),
                         Optional.ofNullable(operatorSummary.getInfo()).map(operatorInfoCodec::toJson),
-                        operatorSummary.getRuntimeStats()))
+                        operatorSummary.getRuntimeStats(),
+                        getPlanNodeEstimateOutputSize(operatorSummary.getPlanNodeId(), estimateMap, planIdNodeMap),
+                        estimateMap.containsKey(operatorSummary.getPlanNodeId()) ? estimateMap.get(operatorSummary.getPlanNodeId()).getOutputRowCount() : NaN))
                 .collect(toImmutableList());
     }
 
+    private double getPlanNodeEstimateOutputSize(PlanNodeId nodeId, Map<PlanNodeId, PlanNodeStatsEstimate> estimateMap, Map<PlanNodeId, PlanNode> planIdNodeMap)
+    {
+        if (!estimateMap.containsKey(nodeId)) {
+            return NaN;
+        }
+        PlanNodeStatsEstimate estimate = estimateMap.get(nodeId);
+        if (planIdNodeMap.containsKey(nodeId)) {
+            return estimate.getOutputSizeInBytes(planIdNodeMap.get(nodeId));
+        }
+        return estimate.getOutputSizeInBytes();
```

Comment:
Do we expect to hit this case? I think not.

Reply:
Should not. Let me refactor this method, and also add an checkArgument statement to make sure the node Id exists in the map and throw better message if not.